src/service.rs

```diff
@@ -62,11 +63,31 @@ pub(crate) enum InnerProtocolHandle {
     Session(Box<dyn SessionProtocol + Send + 'static>),
 }
 
+struct SessionController {
```

Comment:
Renaming is acceptable, but moving to `service.rs` is not a good idea, if need, it should also be moved to `control.rs`, or stay origin

Reply:
Agree. I will move it back `context.rs`.